Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,233 @@
-﻿namespace BootCamp.Chapter
+﻿using Microsoft.VisualBasic.CompilerServices;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        static string defaultReturn = "N/A.";
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances != null && peopleAndBalances.Length != 0)
+            {
+                var highNames = new List<string>();
+                int highBal = 0;
+
+                foreach (var str in peopleAndBalances)
+                {
+                    var arr = str.Split(", ");
+                    var name = arr[0];
+                    int balance = 0;
+                    for (int i = 1; i < arr.Length; i++)
+                    {
+                        var ok = int.TryParse(arr[i], out int number);
+                        if (ok)
+                            balance = Math.Max(balance, number);
+                    }
+                    if (balance > highBal)
+                    {
+                        highNames.Clear();
+                        highNames.Add(name);
+                        highBal = balance;
+                    }
+                    else if (balance == highBal)
+                    {
+                        highNames.Add(name);
+                    }
+                }
+
+                StringBuilder sb = new StringBuilder();
+                var highBalStr = FormatValue(highBal);
+
+                BuildNamesString(sb, highNames);
```

Comment:
Slightly mixed responsibilities. Instead of passing a string builder inside that method- make SB just a detail. Return string instead. So:
instead of:
```cs
StringBuilder sb = new StringBuilder();
                var highBalStr = FormatValue(highBal);

                BuildNamesString(sb, highNames);
                sb.Append($" had the most money ever. {highBalStr}.");

                return sb.ToString();
```
```cs
return BuildNames(highBalstr, highNames);
```

Reply:
Move the selected part to BuildNamesString method, because that's what it does

Reply:
How can I account for the variable strings within the method? (i.e. "had the most money ever", "had the least money ever", etc). Should I pass this as another parameter? Same goes for the pluralOrSingle logic.

Reply:
Yes, pass them as in my example. It's better than passing StringBuilder. StringBuilder is a detail- it's optional for the overall working of the function. However the essence of that function is work with those two strings variables
\

Reply:
Ok. I've moved everything into the BuildNames method.